src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningConditionMapper.java

```diff
@@ -0,0 +1,367 @@
+            if (virtualPk.getRangeKey().isEmpty()) {
+                return getRangeKeyEquals(request, keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue));
+            }
+
+            // "#vhk = :vhk", has virtual RK ==> "#prk BETWEEN vhk AND :vhk+FFFFFF"
+            if (virtualRkCondition.isEmpty()) {
+                AttributeValue lower = keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, EMPTY_BYTE_ARRAY);
+                AttributeValue upper = keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, EMPTY_BYTE_ARRAY, true);
+                return getRangeKeyBetween(request, lower, upper);
+            }
+
+            ComparisonOperator comparisonOperator = virtualRkCondition.get().operator;
+            List<AttributeValue> vrkValues = virtualRkCondition.get().values;
+
+            // "#vhk = :vhk and #vrk = :vrk" ==> "#prk = :vhk+vrk“
+            if (comparisonOperator == EQ) {
+                AttributeValue value = keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, vrkValues.get(0));
+                return getRangeKeyEquals(request, value);
+            }
+
+            BetweenRange betweenRange = getBetweenRange(comparisonOperator, vrkValues);
+            return getRangeKeyBetween(request, betweenRange.lower, betweenRange.upper);
+        }
+
+        private BetweenRange getBetweenRange(ComparisonOperator comparisonOperator, List<AttributeValue> vrkValues) {
+            ScalarAttributeType virtualRkType = virtualPk.getRangeKeyType().get();
+            switch (comparisonOperator) {
+                case LT:
+                    // "#vhk = :vhk and #vrk < :vrk" ==> "#prk BETWEEN :vhk AND :vhk+(vrk-1)+FFFFFF“
+                    ByteBuffer rkBytes = PhysicalRangeKeyBytesConverter.toBytes(virtualRkType, vrkValues.get(0));
+                    byte[] rkByteArray = Arrays.copyOf(rkBytes.array(), rkBytes.limit());
+                    subtractOneFromByteArray(rkByteArray);
+                    return new BetweenRange(
+                        keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, EMPTY_BYTE_ARRAY),
+                        keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, rkByteArray, true));
+                case LE:
+                    // "#vhk = :vhk and #vrk <= :vrk" ==> "#prk BETWEEN :vhk AND :vhk+vrk“
+                    return new BetweenRange(
+                        keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, EMPTY_BYTE_ARRAY),
+                        keyMapper.toPhysicalRangeKey(virtualPk, virtualHkValue, vrkValues.get(0)));
+                case GT:
+                    // #vhk = :vhk and #vrk > :vrk" ==> "#prk BETWEEN :vhk+vrk+0 AND :vhk+FFFFFF“
+                    rkBytes = PhysicalRangeKeyBytesConverter.toBytes(virtualRkType, vrkValues.get(0));
+                    byte[] lowerBoundBytes = Arrays.copyOf(rkBytes.array(), rkBytes.limit() + 1);
+                    lowerBoundBytes[lowerBoundBytes.length - 1] = 0;
```

Comment:
What if the vhk+vrk already maxes out the prk length?

Reply:
Good point -- should check for that and handle it differently. If vhk+vrk is already at vhk+FFFFFF, then throw; otherwise, lowerBound is vhk+(vrk+1).

Reply:
Fixing this now. I now have this such that if the length of vhk+vrk is at the maximum capacity, then find the least signficant i such that vrk[i] is less than the max unsigned byte. Add 1 to vrk[i], and return vrk[0, i].